File: frontend/src/features/mde/constants/riskThresholds.js
export const RISK_THRESHOLD_BANDS = {
  low_upto_pct_rank: 0.2,
  medium_upto_pct_rank: 0.4,
  high_upto_pct_rank: 0.6,
  critical_above_pct_rank: 0.8,
  suspicious_above_pct_rank: 0.5,
}

export const RISK_LEVEL_STYLES = {
  Critical: 'shadow-[inset_0_0_0_1px_rgba(244,63,94,0.35)] bg-rose-500/5',
  High: 'shadow-[inset_0_0_0_1px_rgba(249,115,22,0.35)] bg-orange-500/5',
  Medium: 'shadow-[inset_0_0_0_1px_rgba(34,211,238,0.22)] bg-cyan-500/5',
  Low: 'shadow-[inset_0_0_0_1px_rgba(148,163,184,0.24)] bg-slate-500/5',
}

export const ALERT_TONE = {
  critical: 'border-rose-300/35 bg-rose-500/10 text-rose-100',
  high: 'border-orange-300/35 bg-orange-500/10 text-orange-100',
  medium: 'border-cyan-300/35 bg-cyan-500/10 text-cyan-100',
  low: 'border-slate-300/30 bg-slate-500/10 text-slate-200',
}

File: backend/app/services/risk_thresholds.py
"""Shared risk thresholds for scoring and reporting."""
from __future__ import annotations

from typing import Final

LOW_UPTO_PCT_RANK: Final[float] = 0.2
MEDIUM_UPTO_PCT_RANK: Final[float] = 0.4
HIGH_UPTO_PCT_RANK: Final[float] = 0.6
CRITICAL_ABOVE_PCT_RANK: Final[float] = 0.8
SUSPICIOUS_ABOVE_PCT_RANK: Final[float] = 0.5

RISK_THRESHOLD_BANDS: Final[dict[str, float]] = {
    "low_upto_pct_rank": LOW_UPTO_PCT_RANK,
    "medium_upto_pct_rank": MEDIUM_UPTO_PCT_RANK,
    "high_upto_pct_rank": HIGH_UPTO_PCT_RANK,
    "critical_above_pct_rank": CRITICAL_ABOVE_PCT_RANK,
    "suspicious_above_pct_rank": SUSPICIOUS_ABOVE_PCT_RANK,
}


def risk_level_for_percentile(percentile: float) -> str:
    if percentile >= CRITICAL_ABOVE_PCT_RANK:
        return "CRITICAL"
    if percentile >= HIGH_UPTO_PCT_RANK:
        return "HIGH"
    if percentile >= MEDIUM_UPTO_PCT_RANK:
        return "MEDIUM"
    return "LOW"


def is_suspicious_percentile(percentile: float) -> bool:
    return percentile >= SUSPICIOUS_ABOVE_PCT_RANK


def is_critical_percentile(percentile: float) -> bool:
    return percentile >= CRITICAL_ABOVE_PCT_RANK
